rgo/builders/copr.py

```diff
@@ -189,26 +217,20 @@ def wait_for_results(self):
 
                 done = True
                 for component in batch:
-                    if component.done:
-                        continue
-
-                    build = self.client.build_proxy.get(component.build_id)
-
-                    component.state = build.state
-                    component.done = build.state not in ("waiting", "forked", "running", "pending", "starting", "importing")
-                    component.success = build.state == "succeeded"
-
-                    if component.done:
-                        if component.success:
-                            LOGGER.info("  build #%d - %s: succeeded", component.build_id, component.name)
-                        else:
-                            # failed/cancelled
-                            LOGGER.info("  build #%d - %s: %s", component.build_id, component.name, component.state)
-                            failed = True
-                    else:
-                        done = False
-                        LOGGER.info("  build #%d - %s: %s", component.build_id, component.name, component.state)
-
+                    if not component.done:
+                        component.build_proxy = self.client.build_proxy.get(component.build_id)
+                        component.done, failed = utils.report_build_proxy_and_get_status(component.build_proxy,
+                                                                                         component.name)
```

Comment:
This function can again be a private method of this class and you can put the proxy `get` call inside it as well.

Reply:
I will make it a private method of the class but I am not sure I like adding the `get` call in there. It would have to take the `component` and `override` as an argument and treat them the same even though they have a different type.